packages/app/src/components/Settings/EditModal.js

```diff
@@ -74,7 +84,17 @@ const EditModal = ({
                         ? new Date(closedOn)
                         : null
                     }
-                    onChange={(value) => handleClosedOnDateChanges(value)}
+                    onChange={(date) => {
+                      if (!date) {
+                        handleClosedOnDateChanges(new Date());
+                      } else {
+                        const yearLength = date.getFullYear().toString()
+                          .length;
+                        if (yearLength === 4) {
+                          handleClosedOnDateChanges(date);
+                        }
+                      }
+                    }}
```

Comment:
Since you're using this in multiple places, maybe it's worth breaking out into its own function, and just saying `onChange=validateAndHandleDate` or something like that?